### PR TITLE
Cleanup elapsed time logging and improve database metadata reporting

### PR DESCRIPTION
**Pull request by joshuacolvin0**
Also add new arb-db tool to handle various database manipulations

### CODE REVIEW

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-avm-cpp/data_storage/include/data_storage/aggregator.hpp [hunk 1]
```diff
@@ -40,8 +40,8 @@ class AggregatorStore {
                    const std::vector<uint256_t>& requests,
                    const uint64_t* log_indexes,
                    const std::vector<char>& data);
+    [[nodiscard]] uint64_t blockCountImpl(const ReadTransaction& tx) const;
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/include/data_storage/aggregator.hpp:43]: no idea, reverted

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 2]
```diff
@@ -263,79 +290,118 @@ InitializeResult ArbCore::initialize(const LoadedExecutable& executable) {
...
         }
     }
 
-    if (coreConfig.database_exit_after) {
-        // Exit program, nothing else to do.  Must return some type of error
-        // so startup is cancelled
-        return {rocksdb::Status::OK(), true};
-    }
-
+    auto reorg_done = false;
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:343]: :+1:

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 3]
```diff
@@ -1723,6 +1755,26 @@ void ArbCore::operator()() {
                         std::cerr << "Error pruning checkpoints: "
                                   << prune_status.ToString() << "\n";
                     }
+                } else {
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1758]: :+1:

**Review by hkalodner — APPROVED** [review]
LGTM

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**